R/mfx-tidiers.R

```diff
@@ -0,0 +1,241 @@
+#'   for various generalised linear models (GLMs). Unlike standard linear models 
+#'   (e.g. OLS regression), estimated model coefficients in a GLM cannot be
+#'   directly interpreted as marginal effects (i.e., the change in the response 
+#'   variable predicted after a one unit change in one of the regressors). This 
+#'   is because the estimated coefficients are multiplicative, dependent on both
+#'   the link function that was used for the estimation and any other variables
+#'   that were included in the model. When calculating marginal effects, users
+#'   must typically choose whether they want to use i) the average observation 
+#'   in the data, or ii) the average of the sample marginal effects. See 
+#'   `vignette("mfxarticle")` from the `mfx` package for more details.
+#'   
+#' @examples
+#' \dontrun{ 
+#' library(mfx)
+#' 
+#' ## Get the marginal effects from a logit regression
+#' mod_logmfx <- logitmfx(am ~ cyl + hp + wt, atmean = T, data = mtcars)
+#' tidy(mod_logmfx, conf.int=T)
+#' 
+#' ## Compare with the naive model coefficients of the same logit call (not run)
+#' # tidy(glm(am ~ cyl + hp + wt, family = binomial, data = mtcars), conf.int=T)
+#' 
+#' augment(mod_logmfx)
+#' glance(mod_logmfx)
+#' 
+#' ## Another example, this time using probit regression
+#' mod_probmfx <- probitmfx(am ~ cyl + hp + wt, atmean = T, data = mtcars)
+#' tidy(mod_probmfx, conf.int=T)
+#' augment(mod_probmfx)
+#' glance(mod_probmfx)
+#' }
+#' 
+#' @family mfx tidiers
+#' @seealso [tidy()], [mfx::betamfx()], [mfx::logitmfx()], [mfx::negbinmfx()], [mfx::poissonmfx()], [mfx::probitmfx()]
+#' @export
+tidy.mfx <-
+  function(x, conf.int = FALSE, conf.level = 0.95, ...) {
+    
+    x_tidy <-
+      x$mfxest %>% 
+      as.data.frame() %>% 
+      tibble::rownames_to_column() %>% 
+      tibble::as_tibble() %>%
+      ## Note: using the ".data" prefix to avoid "no visible binding for global 
+      ## variable" warning during devtools::check()
+      dplyr::select(
+        term=.data$rowname, estimate=.data$`dF/dx`, std.error=.data$`Std. Err.`, 
+        statistic=.data$z, p.value=.data$`P>|z|`
+        )
+    
+    ## Optional: Add "atmean" column
+    ## If no "atmean" argument is specified in the model call, then will default to TRUE
+    if (!grepl("atmean", format(c(x$call)))) {
+      x_tidy$atmean <- TRUE
+      ## Else, extract the user-specified "atmean" argument from the model call
+    } else {
+      x_tidy$atmean <- as.logical(gsub(".*atmean = |,.*|).*", "", format(c(x$call))))
+    }
+    
+    if (conf.int) {
+      x_tidy <-
+        x_tidy %>%
+        dplyr::mutate(
+          conf.low = estimate - qt(1-(1-conf.level)/2, df=x$fit$df.residual)*std.error,
+          conf.high = estimate + qt(1-(1-conf.level)/2, df=x$fit$df.residual)*std.error
+        )
+    }
+    
+    x_tidy <-
+      x_tidy %>%
+      dplyr::select(term, contains("atmean"), everything())
+    
+    x_tidy
+  }
+
+
+#' @rdname tidy.mfx
```

Comment:
Putting all the classes here in this same file is fine. If the code is exactly the same across all model classes, I'm down with this. When the you have to start special case-ing code for different classes is when I like to split things apart.